File: convertDict.py

def htmlEntities(str):
    return str.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;').replace('"', '&quot;').replace("'",'&#39');

import sys
if len(sys.argv) < 2:
    print('Thieu data')
    exit()


dictFile=sys.argv[1]
dictName=dictFile.split('.')[0].encode()
quote = b'"'
stored = b'const ' + dictName + b' = {\n'
endstored = b'};\n'

f = open(dictFile, 'rb').read()
lines = f.splitlines()
for line in lines:
    stored += quote + line.split(b'=', 1)[0] + quote + b': ' + quote + line.split(b'=', 1)[-1] + quote + b','
stored = stored.strip(b',') #loại bỏ dấu phẩy cuối cùng để không phá vỡ quy tắc của object trong JS
final = stored + endstored

fw=open(dictFile+'.pyjs','wb')
fw.write(final)
fw.close()

File: convertDict.js
// Convert tuw VietPhrase format qua JS dictionary
// Chay tu dong lenh. Cu Phap
// node ./convertDict.js ./VietPhrase.txt

function htmlEntities(str) {
    return String(str).replaceAll('&', '&amp;').replaceAll('<', '&lt;').replaceAll('>', '&gt;').replaceAll('"', '&quot;').replaceAll("'",'&#39');
}

let dataFile = process.argv.slice(-1)[0];
if (__filename == dataFile) { 
    console.log('Thieu file data');
    process.exit(1); }


const fs = require('fs');
const dataFileContents = fs.readFileSync(dataFile);

let lines = dataFileContents.toString().trim().split('\r\n');
for (let i=0; i<lines.length-1; i++) lines[i]='"'+htmlEntities(lines[i].trim()).replace('=','":"') + '"';

let dictName = dataFile.split('/'); // split('\) neu o Windows
let outFileContent = ' const ' + dictName[dictName.length-1].split('.')[0]+' = {' + lines.join(',\r\n') + '}; \r\n';

fs.writeFileSync(dataFile + '.js',outFileContent);
